### PR TITLE
Make use of variable "from" in local_vector find() function.

### PR DESCRIPTION
**Pull request by RicardRC**

### CODE REVIEW

**Comment by RicardRC** [comment]
Fixed compilation error.

File: core/templates/local_vector.h [hunk 1]
```diff
@@ -178,7 +178,8 @@ class LocalVector {
 	}
 
 	int64_t find(const T &p_val, U p_from = 0) const {
-		for (U i = 0; i < count; i++) {
+		ERR_FAIL_INDEX_V(p_from, count, -1);
```
Comment (akien-mga) [inline core/templates/local_vector.h:181]: Shouldn't it be `ERR_FAIL_UNSIGNED_INDEX_V`  like for other checks in this class? I'm surprised that it's not raising a warning about tautological comparison to 0 though.

Comment (RicardRC) [reply core/templates/local_vector.h:181]: U is default defined as unsigned, but I don't see how it's enforced to be unsigned if it's defined signed specifically :S

Comment (RicardRC) [reply core/templates/local_vector.h:181]: But I'll change it, it does not make sense to make this check only on one of the functions, though I'd expect the compilers to actually optimize that check out in the cases in which it is tautological.

**Comment by akien-mga** [comment]
Could you squash the two commits into one? See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase).

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
I'm getting this error at runtime when opening any project:
```
ERROR: Index p_from = 0 is out of bounds (count = 0).
   at: find (./core/templates/local_vector.h:181)
```

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.

**Comment by akien-mga** [comment]
Cherry-picked for 3.3.3.

**Pull request merged**